carbonserver/carbonserver/api/routers/authenticate.py

```diff
@@ -29,3 +44,129 @@ def auth_user(
         status_code=status.HTTP_401_UNAUTHORIZED,
         detail="Incorrect password or email!",
     )
+
+
+fief = FiefAsync(
+    settings.fief_url, settings.fief_client_id, settings.fief_client_secret
+)
+
+
+@dataclass
+class FullUser:
+    db_user: dict
+    auth_user: dict
+
+
+SESSION_COOKIE_NAME = "user_session"
+scheme = OAuth2AuthorizationCodeBearer(
+    settings.fief_url + "/authorize",
+    settings.fief_url + "/api/token",
+    scopes={"openid": "openid", "offline_access": "offline_access"},
+    auto_error=False,
+)
+web_scheme = APIKeyCookie(name=SESSION_COOKIE_NAME, auto_error=False)
+auth = fief_auth = FiefAuth(fief, scheme)
+web_auth = FiefAuth(fief, web_scheme)
+api_auth = Annotated[HTTPAuthorizationCredentials, Depends(HTTPBearer())]
+
+
+class UserOrRedirectAuth(FiefAuth):
```

Comment:
more fief & auth specific  than generic user management but otherwise I'm inclined to agree
That being said I don't have a clear view of the "deeper business logic" yet so  I'd rather keep it all in the same place for now and refactor after adding the API token checks & cleared up a few registration/auth questions

Reply:
ok !